packages/flutter_test/lib/src/binding.dart

```diff
@@ -275,6 +275,17 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
     return TestDefaultBinaryMessenger(super.createBinaryMessenger());
   }
 
+  /// For using the default implementaion of [BinaryMessenger] instead of
+  /// test version.
+  ///
+  /// This method is useful for integration tests which uses real platform
+  /// messages such as e2e.
+  ///
+  /// See: https://pub.dev/packages/e2e
+  BinaryMessenger createRealBinaryMessenger() {
```

Comment:
Is this the right binding for e2e testing? Should we use something that doesn't mock stuff out? For example, this one also mocks out network. What do you think of having a specialized `E2eWidgetsFlutterBinding` (provided by `package:e2e`) that provides the exact implementation of the binary messenger it needs. See, for example, [_DriverBinding](https://github.com/flutter/flutter/blob/bf8e2c1449423a6f2e00919f8ee329f742390cd9/packages/flutter_driver/lib/src/extension/extension.dart#L45) provided by `package:flutter_driver` for driver tests.

/cc @Hixie who IIRC designed these bindings.

Reply:
e2e is on top of LiveTestWidgetsFlutterBinding. So there is one more parent in between.

Reply:
Ah, at the time of writing my comment I didn't know that you were already adding `E2eWidgetsFlutterBinding`. I guess the issue might be that `TestFlutterWidgetsBinding` does more than just support `WidgetTester`. Perhaps we should reduce the scope of `TestFlutterWidgetsBinding` to not provide opinion beyond what `WidgetTester` needs, and have its subclasses add those opinions. For example, `TestFlutterWidgetsBinding` shouldn't override `createBinaryMessenger` and shouldn't mock out `HttpClient`. Then `E2eWidgetsFlutterBinding`, which is designed to run in a real non-mock environment, could use real implementations of those things. `flutter test`, OTOH, would mock stuff out more aggressively.